### PR TITLE
Lack of actual initial JSON-RPC API

### PR DESCRIPTION
**Pull request by dulanov**
Related to #105.

### CODE REVIEW

File: docs/api.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    * addr - address of created keyfile
+
+== emerald_importKeyfile(data, passphrase)
+
+Import previously created keystore file into connector.
+
+Arguments:
+
+    * data - keyfile to be imported into connector
```
Comment (sorpaas) [inline docs/api.adoc]: @dulanov Is this parameter the complete private key or a path to a file that contains the key?

File: docs/api.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    * addr - address of imported keyfile
+
+== emerald_signTransaction(tr, passphrase)
+
+Signs transaction offline with private key from keystore file by given passphrase.
+
+Arguments:
+
+    * tr - given transaction
```
Comment (sorpaas) [inline docs/api.adoc]: What is the struct for `tr`? Is it the same as Ethereum RPC like below?

```
pub struct RPCTransaction {
    pub hash: String,
    pub nonce: String,
    pub blockHash: String,
    pub blockNumber: String,
    pub transactionIndex: String,
    pub from: String,
    pub to: String,
    pub value: String,
    pub gas: String,
    pub gasPrice: String,
    pub input: String,
}
```

**Comment by sorpaas** [comment]
@dulanov Besides questions I listed in the code comments, another one is about bytearray encoding. So we'll use the same approach as Ethereum RPC and encode bytearray as hex strings, right?

Right now there's a single difference for Ethereum RPC encoding and the one SputnikVM accepts -- SputnikVM doesn't require hex strings to be prefixed by `0x`. What's the case for emerald? Do you think it's a good idea to require `0x` prefixing or make it optional?

File: Cargo.toml [hunk 3]
```diff
@@ -1,12 +1,12 @@
 [package]
 name = "emerald"
 version = "0.0.1"
-authors = ["Dmitry Ulanov <dulanov@gmail.com>"]
-description = "Ethereum classic web3 like connector written in Rust"
+authors = ["Dmitry Ulanov <dulanov@gmail.com>", "Constantine Kryvomaz <kostiyantynk@gmail.com>"]
+description = "Ethereum classic offline wallet written in Rust"
```
Comment (sorpaas) [inline Cargo.toml]: s/classic/Classic

File: docs/api.adoc (hunk 1 above)
Comment (dulanov) [inline docs/api.adoc]: @sorpaas Keyfile compelete content encoded in JSON.

File: README.adoc [hunk 4]
```diff
@@ -13,11 +13,11 @@ ifdef::badges[]
 = Emerald-rs image:https://img.shields.io/travis/ethereumproject/emerald-rs/master.svg?style=flat-square["Build Status", link="https://travis-ci.org/ethereumproject/emerald-rs"] image:https://img.shields.io/appveyor/ci/dulanov/emerald-rs/master.svg?style=flat-square["Build Status", link="https://ci.appveyor.com/project/dulanov/emerald-rs"] image:https://img.shields.io/badge/License-Apache%202.0-blue.svg?style=flat-square&maxAge=2592000["License", link="https://github.com/ethereumproject/emerald-rs/blob/master/LICENSE"]
 endif::[]
 
-Ethereum classic web3 like connector written in Rust.
+Ethereum classic offline wallet written in Rust.
```
Comment (sorpaas) [inline README.adoc]: s/classic/Classic/

Is the non-cap version of `classic` intentional?

File: docs/api.adoc (hunk 2 above)
Comment (dulanov) [inline docs/api.adoc]: Yes, right, I am working futher, it was just a draft...

**Comment by dulanov** [comment]
@sorpaas yes, generally we should implement something like `personal` module from geth, but with some additions - https://github.com/ethereum/go-ethereum/wiki/Management-APIs#personal.

**Comment by dulanov** [comment]
About `0x` prefix, for emerald it's optional now, you can add it or not - no difference

File: docs/api.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    * version - the current version
+
+== emerald_newAccount(name, desc, passphrase)
+
+Creates new keystore file and caches it.
+
+Arguments:
+
+    * name - name for account (optional)
```
Comment (sorpaas) [inline docs/api.adoc]: Can we put optional parameters in the end? It seems like JSON RPC uses an array to represent all parameters, if I understand this correct.

Putting optional ones to the end would avoid unnecessary confusion and potential code failures (for example, get an account description as the passphrase).

Comment (dulanov) [inline docs/api.adoc]: Yes yes, I just sarted, need one more day, sorry for that

File: README.adoc [hunk 6]
```diff
@@ -83,16 +68,16 @@ fn main() {
         .parse::<SocketAddr>()
         .expect("Expect to parse address");
 
-    let client_addr = "127.0.0.1:8545"
-        .parse::<SocketAddr>()
-        .expect("Expect to parse client address");
-
-    emerald::start(&addr, &client_addr, None);
+    emerald::start(&addr, None);
```
Comment (sorpaas) [inline README.adoc]: Emerald also needs the `base_path` for finding the place to store keyfiles, etc. Maybe there should be an extra parameter?

Comment (dulanov) [inline README.adoc]: It just draft, modify how do think will be better.

Comment (sorpaas) [inline README.adoc]: s/`emerald::start(&addr, None);`/`emerald::start(&addr, None, None);`/

That would fit the current fn definition.

Comment (dulanov) [inline README.adoc]: Thank you.

**Pull request merged**